src/adm.rs

```diff
@@ -54,24 +284,11 @@ pub async fn get_tiles(
         .error_for_status()?
         .json()
         .await?;
+    // TODO: Is there a better way to raise the error possibly inside of `filter_and_process`?
     response.tiles = response
         .tiles
         .into_iter()
-        .filter_map(filter_and_process)
+        .filter_map(|tile| state.filter.filter_and_process(tile, tags))
```

Comment:
We might want to add a `.take(settings.MAX_TILES)` here to cap the return tile set. Firefox is anticipating no more than 2 sponsored tiles for now.